kubeflow/trainer/rhai/traininghub.py

```
@@ -332,6 +334,44 @@ def _read_sft_metrics(self):
+            LoRA writes a single metrics file (no per-rank wildcard).
+            Each line is a JSON object with step, epoch, loss, learning_rate.
+            """
+            metrics_file = f"{ckpt_output_dir}/{metrics_file_rank0}"
+
+            try:
+                if not os.path.exists(metrics_file):
+                    return {}
+
+                # Read last line of metrics using tail
+                try:
+                    result = subprocess.run(
```

Ruff (0.15.6)

[error] `subprocess` call: check for execution of untrusted input

(S603)